File: Tests/Scraping/MessageFolderScrapingTests.m
//  MessageFolderScrapingTests.m
//
//  Copyright 2013 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

#import "AwfulScrapingTestCase.h"
#import "AwfulPrivateMessageFolderScraper.h"

@interface MessageFolderScrapingTests : AwfulScrapingTestCase

@end

@implementation MessageFolderScrapingTests

+ (Class)scraperClass
{
    return [AwfulPrivateMessageFolderScraper class];
}

- (void)testInbox
{
    AwfulPrivateMessageFolderScraper *scraper = [self scrapeFixtureNamed:@"private-list"];
    NSArray *messages = scraper.messages;
    XCTAssertEqual(messages.count, (NSUInteger)4);
    NSArray *allMessages = [AwfulPrivateMessage fetchAllInManagedObjectContext:self.managedObjectContext];
    XCTAssertEqual(messages.count, allMessages.count);
    NSArray *allUsers = [AwfulUser fetchAllInManagedObjectContext:self.managedObjectContext];
    XCTAssertEqual(allUsers.count, (NSUInteger)3);
    
    AwfulPrivateMessage *tagged = [AwfulPrivateMessage fetchArbitraryInManagedObjectContext:self.managedObjectContext
                                                                    matchingPredicateFormat:@"from.username = 'CamH'"];
    XCTAssertEqualObjects(tagged.messageID, @"4549686");
    XCTAssertEqualObjects(tagged.subject, @"Re: Awful app etc.");
    XCTAssertEqual(tagged.sentDate.timeIntervalSince1970, 1348778940.);
    XCTAssertEqualObjects(tagged.threadTag.imageName, @"sex");
    XCTAssertTrue(tagged.replied);
    XCTAssertTrue(tagged.seen);
    XCTAssertFalse(tagged.forwarded);
}

@end

File: Tests/Scraping/MessageFolderScrapingTests.swift
//  MessageFolderScrapingTests.swift
//
//  Copyright 2013 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

import XCTest
import Awful

class MessageFolderScrapingTests: ScrapingTestCase {
    override class func scraperClass() -> AnyClass {
        return PrivateMessageFolderScraper.self
    }

    func testInbox() {
        let scraper = scrapeFixtureNamed("private-list") as PrivateMessageFolderScraper
        let messages = scraper.messages
        XCTAssertTrue(messages.count == 4)
        XCTAssertEqual(messages.count, fetchAll(PrivateMessage.self, inContext: managedObjectContext).count)
        
        let tagged = fetchOne(PrivateMessage.self, inContext: managedObjectContext, matchingPredicate: NSPredicate(format: "rawFromUsername = 'CamH'"))!
        XCTAssertEqual(tagged.messageID, "4549686")
        XCTAssertEqual(tagged.subject!, "Re: Awful app etc.")
        XCTAssertEqual(tagged.sentDate!.timeIntervalSince1970, 1348778940)
        XCTAssertEqual(tagged.threadTag!.imageName!, "sex")
        XCTAssertTrue(tagged.replied)
        XCTAssertTrue(tagged.seen)
        XCTAssertFalse(tagged.forwarded)
    }
}
